servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/DefaultLoadBalancer.java

```diff
@@ -502,6 +509,24 @@ public Completable closeAsyncGracefully() {
 
     @Override
     public List<Entry<ResolvedAddress, List<C>>> usedAddresses() {
```

Comment:
I may be missing something here, but isn't that a check-then-act issue? What guarantees that the condition still holds after the IF.